eth: update contract bindings and code cleanup

yondonfu opened the pull request:
**What does this pull request do? Explain your changes. (required)**
<!-- A clear and concise description of what this pull request does. -->

Update contract bindings to reflect the latest changes in the protocol contracts [pm](https://github.com/livepeer/protocol/tree/pm) branch. In preparation for integrating the `pm` package with the rest of the go-livepeer codebase and for interactions with the updated contracts, this PR seeks to remove all remaining V1 protocol job related code.

**Specific updates (required)**
<!--- List out all significant updates your code introduces -->
- Introduces LivepeerETHTicketBroker contract binding
- Updates the rest of the contract bindings
- Remove job related code

**How did you test each of these updates (required)**
<!-- A detailed description of how you tested your code changes. Include details of your testing environment, and the tests you ran to see how your change affects other areas of the code, etc. -->

I built the node and ran it in broadcaster/orchestrator mode in devenv

**Does this pull request close any open issues?**
<!-- Fixes # -->

Fixes #652 

**Checklist:**
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] README and other documentation updated
- [x] Node runs in OSX and devenv
- [x] All tests in `./test.sh` pass

yondonfu committed:
eth: update contract bindings

- Introduces LivepeerETHTicketBroker
- Code cleanup: Remove job related code

eladmallel commented on server/mediaserver.go:
```diff
@@ -277,23 +277,8 @@ func gotRTMPStreamHandler(s *LivepeerServer) func(url *url.URL, rtmpStrm stream.
-				}
-				return ErrBroadcast
-			}
+			// TODO: Check broadcaster's deposit with TicketBroker
```
Perhaps worth capturing as an issue? - I'm concerned we might miss this. wdyt?

eladmallel reviewed (approved):
Besides the one little comment I added, LGTM. However, I'm probably not the best person to review this kind of change.. wondering if perhaps another set of eyes should review this PR?

yondonfu committed:
Merge branch 'master' into yf/update-contract-bindings

yondonfu committed:
Fix build

- Put back missing import in common/db_test.go

yondonfu commented:
Since master is known to be unstable and in the interest of avoiding further merge conflicts with other people working off of master I'd be ok with getting this merged sooner rather than later. I think it is also worth noting that master as is also won't work with the `pm` branch of the protocol repo.

Putting the merge on hold because I'm going to push an update to this PR to make sure LP client adheres to the `pm/Broker` interface

yondonfu committed:
eth + pm: client implements Broker

- Use a single getter for deposit/penalty escrow in Broker
- Add transaction to return signature of Broker functions
- Add wrapper functions to client to implement Broker

yondonfu commented:
661f2c9: client implements Broker

Tested by adding the following line in `client.go` which should raise a compiler error if the client did not implement Broker:

```
var _ pm.Broker = client{}
```

j0sh commented on cmd/livepeer_cli/wizard_stats.go:
```diff
@@ -325,14 +315,6 @@ func (w *wizard) getEthBalance() string {
 }
 
-func (w *wizard) getDeposit() string {
-	e := httpGet(fmt.Sprintf("http://%v:%v/broadcasterDeposit", w.host, w.httpPort))
```
Don't we still need a deposit?

yondonfu replied:
Yeah, but it is with a different contract so this PR cleaned up the old code and the new code for the new contract is included in #658 which fetches all of a PM sender's information in a single request instead of individually fetching the sender's deposit and penalty escrow.
